### PR TITLE
feat: render code blocks in markdown

### PR DESCRIPTION
**Pull request by Hellebore**
- Render markdown response from AI into html, apply syntax highlighting and sanitize
- Remaining issues are the scrollbar not being right and handling theme changes - will handle these separately

## Checklist

- [ ] If `package.json` or `yarn.lock` have changed, then test the VSIX built by `yarn run vsce package` works from a direct install

### CODE REVIEW

File: src/chat.ts [hunk 1]
```diff
@@ -36,24 +68,40 @@ export class ChatProvider implements vscode.WebviewViewProvider {
...
   }
 
-  public addResult(result) {
+  public addResult(result: ChatResult) {
+    // Send the whole message we've been streamed so far to the webview,
+    // after converting from markdown to html
+
+    if (result.outcome === ChatResultOutcome.Success) {
+      this.currentAssistantMessage += result.textContent;
+    }
```
Comment (Hellebore) [inline src/chat.ts:88]: build up the whole current message here so we can convert the whole thing to html and send it each time we get a streaming response

File: src/extension.ts [hunk 2]
```diff
@@ -383,41 +383,47 @@ function registerCommands(
...
-        activeFile = activeEditor.document.uri;
-      }
-      const allFiles = [];
-      for (const tabGroup of vscode.window.tabGroups.all) {
-        for (const tab of tabGroup.tabs) {
-          if (tab.input instanceof vscode.TabInputText) {
-            allFiles.push(tab.input.uri);
+    commands.registerCommand(
+      "sourcery.chat_request",
+      (message: ChatRequest) => {
```
Comment (Hellebore) [inline src/extension.ts:388]: added in some typing which seems to have changed formatting

File: package.json [hunk 3]
```diff
@@ -398,9 +398,10 @@
 		"vscode": "npm run vscode:prepublish && VSCODE=$(which code-insiders || which code || echo echo ERROR: neither the code nor code-insiders vscode executable is installed); USER=dummy-dont-share-vscode-instance \"$VSCODE\" --user-data-dir=$PWD/.vscode-dev/user-data --verbose --extensionHomePath=$PWD/.vscode-dev/extensions --extensionDevelopmentPath=$PWD $*",
 		"vscode_local": "SOURCERY_EXECUTABLE=../core/run-sourcery.sh yarn run vscode",
 		"format": "prettier --write .",
-  		"lint:format": "prettier --check ."
+		"lint:format": "prettier --check ."
 	},
 	"devDependencies": {
+		"@types/marked": "^5.0.0",
```
Comment (brendanator) [inline package.json:404]: **🎯 suggestion:** ‏Increase the node version in the build scripts

Marked didn't install on my machine until I upgraded from node version 14 to the latest

**Review by brendanator — APPROVED** [review]
It's looking nice!

Here's a few things I've spotted:
- Bullet points appear outside the chat bubble, and headers seems pretty big
image
- When generating code wider than the sidebar it scrolls to the right 
image
- I seem to have different themes in different code blocks - check the background colours
image
- Is it possible to use the theme of vscode? Mine chat code looks totally different from the editor 
image
- Just code on it's own looks a little funny
image
- I'd expect when I click a recipe button that it displays as a user message rather than a assistant message. e.g. `Run recipe - generate unit tests`
image

**Comment by Hellebore** [comment]
> It's looking nice!
> 
> Here's a few things I've spotted:
> 
> * Bullet points appear outside the chat bubble, and headers seems pretty big
>   image
> * When generating code wider than the sidebar it scrolls to the right
>   image
> * I seem to have different themes in different code blocks - check the background colours
>   image
> * Is it possible to use the theme of vscode? Mine chat code looks totally different from the editor
>   image
> * Just code on it's own looks a little funny
>   image
> * I'd expect when I click a recipe button that it displays as a user message rather than a assistant message. e.g. `Run recipe - generate unit tests`
>   image

Good points - have created a new issue for these

**Pull request merged**